RUC-MSc-CS-CIT-2024/portfolio_subproject_2
# feat: FrameworkContext & Models

- Used OnConfiguring to load the connection string from environment variables.
- Defined models (User, Bookmark, Completed, SearchHistory, UserScore, Following) aligned with database schema.
- Applied metadata annotations (e.g., [Key], [Required], [Column]) to specify primary keys, required fields, and column mappings.
- Configured appsettings.json to use placeholders for environment variables, referencing .env for DB details.
- Added .env to .gitignore to secure sensitive data, and instructed each team member to create a local .env with their DB credentials (DB_HOST, DB_NAME, DB_USER, DB_PASSWORD).

## How to test it will work for you guys aswell
Create .env inside the API project Populate .env with your own DB credentials.
Run the Application: Confirm the connection by starting the app; it will load the DB connection from .env via appsettings.json. (Here you can run the API project with dotnet run in the terminal, then navigate to http://localhost:5187/test-connection ... and it should give you a "Connection successful! User count: 4" response)
